lib/SerializedCache.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Amp\Cache;
+
+use Amp\Promise;
+
+interface SerializedCache extends Cache
+{
+    /**
+     * The promise is resolved with the unserialized cached value.
+     *
+     * @param $key string Cache key.
+     *
+     * @return Promise<mixed|null> Resolves to the cached value nor `null` if it doesn't exist or fails with a
+     * CacheException on failure.
+     */
+    public function get(string $key): Promise;
```

Comment:
`SerializedCache` can't extend `Cache`, because the return type is widened here from `Promise<string|null>` to `Promise<mixed|null>`.